lib/types.js

```diff
@@ -152,7 +152,13 @@ module.exports = function(expect) {
           }
         }
       : Object.keys,
-    equal(a, b, equal) {
+    equal(a, b, equal, equalityTypes) {
+      // default to the type itself in the absence of specific type information
+      equalityTypes = equalityTypes || {
+        aType: this,
+        bType: this
+      };
```

Comment:
This seems like the wrong place to default these values, couldn't this be handled by the caller.

Reply:
@sunesimonsen I definitely agree, but in this case I was very concerned there might be external plugins that have made assumptions that would case calling `.equal()` to break. That said, it may well be that all the places that _set_ these values are actually in core (I confess it's been a little tricky to follow all the calls to equality).

Reply:
I think the equal calls on the types will always be called by core only, but I could be mistaken.